design/userdefined-rootdefinehints.md

```diff
@@ -0,0 +1,103 @@
+<!--
+ This work is licensed under a Creative Commons Attribution 3.0
+ Unported License.
+
+ http://creativecommons.org/licenses/by/3.0/legalcode
+-->
+
+# userdefined-rootdevicehints
+
+## Status
+
+provisional
+
+## Table of Contents
+
+<!--ts-->
+   * [userdefined-rootdevicehints](#userdefined-rootdevicehints)
+      * [Status](#status)
+      * [Table of Contents](#table-of-contents)
+      * [Summary](#summary)
+      * [Motivation](#motivation)
+         * [Goals](#goals)
+         * [Non-Goals](#non-goals)
+      * [Proposal](#proposal)
+         * [Implementation Details/Notes/Constraints](#implementation-detailsnotesconstraints)
+         * [Risks and Mitigations](#risks-and-mitigations)
+      * [Design Details](#design-details)
+         * [Work Items](#work-items)
+         * [Dependencies](#dependencies)
+         * [Test Plan](#test-plan)
+         * [Upgrade / Downgrade Strategy](#upgrade--downgrade-strategy)
+         * [Version Skew Strategy](#version-skew-strategy)
+      * [References](#references)
+
+<!-- Added by: mikkosest -->
+
+<!--te-->
+
+## Summary
+
+This document explains how rootdevicehints could be given as user defined parameters as part BaremetalHostSpec. The idea is that user would have a possibilty to defines selective constrains for root device selection. Sometimes selective constrains will not be completely deterministic, but it is upto user, which hints to use.
+
+## Motivation
+
+We need to be able select root device, when there is multiple devices available so that device is not chosen randomly
+
+### Goals
+
+1. Decide root device hints are needed be supported.
+2. Agree on the way haw information is propagated to ironic through BMO
+
+
+### Non-Goals
+
+The complete set of root device hints as input is not needed only those seen important.
+
+## Proposal
+
+### Implementation Details/Notes/Constraints
+
+- BaremetalHostSpec need to be updated to support list of following root device hints:
+* name
```

Comment:
Is "name" going to be a clear enough description for the field? Maybe it is in the context of the full data structure. What does the proposed struct look like?

Reply:
Below complete proposed struct, after decision in this doc,  it will be a subset of that

type RootDeviceHints struct {
	// A device name like "/dev/vda"
	DeviceName string `json:"devicename,omitempty"`

	// A SCSI bus address like 0:0:0:0
	HCTL string `json:"hctl,omitempty"`

	// Device identifier
	Model string `json:"model,omitempty"`

	// Device vendor
	Vendor string `json:"vendor,omitempty"`

	// Disk serial number
	Serial string `json:"serial,omitempty"`

	// Size of the device in GiB
	Size int `json:"size,omitempty"`

	// Unique storage identifier
	WWN string `json:"wwn,omitempty"`

	// Unique storage identifier with the vendor extension appended
	WWN_with_extention string `json:"wwn_with_extention,omitempty"`

	// Uunique vendor storage identifier
	WWN_vendor_extention string `json:"wwn_vendor_extention,omitempty"`

	// Device rotational type
	Rotational string `json:"rotational,omitempty"`
}

Reply:
Should we incorporate the suggested struct into this doc?

Also a nit, we shouldn't do snake case/underscores for variable names. Everything needs to be camel case, e.g. WWNWithExtension

Reply:
I added struct as a part of doc

Reply:
I would like to provide size in gigabytes to avoid conversion from bytes to GiB. Is that ok or should we try to keep the same unit as In legacy Inspect bytes are used?